lib/spack/spack/version.py

```diff
@@ -323,13 +315,9 @@ def satisfies(self, other):
         gcc@4.7 so that when a user asks to build with gcc@4.7, we can find
         a suitable compiler.
         """
-        self_cmp = self._cmp(other.commit_lookup)
-        other_cmp = other._cmp(self.commit_lookup)
```

Comment:
It doesn't look like there is an implementation of `up_to` in `GitVersion` that preserves this behavior -- should there be?